### PR TITLE
net: http client: implement high level API of http client

### PR DESCRIPTION
**Pull request by chinli**
- revert the net http client, and modified some definition.

Signed-off-by: Frank Lee <frank.lee@unisoc.com>

### CODE REVIEW

File: include/net/http_client.h [hunk 1]
```diff
@@ -1,33 +1,20 @@
...
-/**
- * @brief Http client library
- * @defgroup http client Library
- * @ingroup networking
- * @{
- */
-
 #ifdef CONFIG_HTTPS
-	#define CONFIG_NET_SOCKETS_SOCKOPT_TLS
+#define CONFIG_NET_SOCKETS_SOCKOPT_TLS
```
Comment (bublover) [inline include/net/http_client.h]: config NET_SOCKETS_SOCKOPT_TLS                                                                                                                                                                          
      bool "Enable TCP TLS socket option support [EXPERIMENTAL]"
      select TLS_CREDENTIALS
     help
        Enable TLS socket option support which automatically establishes
        a TLS connection to the remote host.

It's not a good way. You should enable it in your proj.conf or Kconfig dependency.

Comment (bublover) [reply include/net/http_client.h]: And, I have no idea about EXPERIMENTAL.

File: subsys/net/lib/http/http_client.c [hunk 2]
```diff
@@ -279,8 +195,9 @@ static int http_client_request(struct http_ctx *ctx)
 	ctx->rsp_buf = malloc(HTTP_RESPONSE_MAX_SIZE);
 	if (ctx->rsp_buf < 0) {
```
Comment (bublover) [inline subsys/net/lib/http/http_client.c]: I don't think it's working here.

File: subsys/net/lib/http/http_client.c [hunk 3]
```diff
@@ -376,8 +375,9 @@ int http_client_get(struct http_ctx *ctx,
 
 	/**add http end */
 	strcat(ctx->req_buf, "\r\n");
-
+	printk("http request size=%d\n", strlen(ctx->req_buf));
```
Comment (bublover) [inline subsys/net/lib/http/http_client.c]: Keep the uniform log print.
NET_DBG is proper here.

File: subsys/net/lib/http/http_client.c (hunk 2 above)
Comment (chinli) [reply subsys/net/lib/http/http_client.c]: could you help tell me why~！which lines?

Comment (bublover) [reply subsys/net/lib/http/http_client.c]: Refer to prototype.
void *malloc(size_t size);
You could get NULL retval if it failed to allocate memory.

Comment (chinli) [reply subsys/net/lib/http/http_client.c]: Got it!

**Pull request merged**